src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+            window->number = stationId;
+            auto station = stationmgr::get(stationId);
+            window->owner = station->owner;
+            window->min_width = 192;
+            window->min_height = 137;
+            window->max_width = 600;
+            window->max_height = 440;
 
-        return (window*)regs.esi;
+            auto skin = objectmgr::get<interface_skin_object>();
+            if (skin != nullptr)
+            {
+                window->colours[1] = skin->colour_0A;
+            }
+            // 0x0048F29F end
+
+            window->saved_view.clear();
+        }
+        // TODO(avgeffen): only needs to be called once.
+        common::initEvents();
+
+        window->current_tab = 0;
+        window->invalidate();
+
+        window->widgets = station::widgets;
+        window->enabled_widgets = station::enabledWidgets;
+        window->holdable_widgets = 0;
+        window->event_handlers = &station::events;
+        window->activated_widgets = 0;
+        window->disabled_widgets = 0;
+        window->init_scroll_widgets();
+        station::initViewport(window);
+
+        return window;
     }
 
-    // 0x0048EF02
-    static void draw_rating_bar(window& w, gfx::drawpixelinfo_t& dpi, int16_t x, int16_t y, uint8_t amount, colour_t colour)
+    namespace cargo
     {
-        registers regs;
-        regs.al = amount;
-        regs.cx = x;
-        regs.dx = y;
-        regs.esi = (int32_t)&w;
-        regs.edi = (int32_t)&dpi;
-        regs.ebp = colour;
-        call(0x0048EF02, regs);
+        enum widx
+        {
+            scrollview = 7,
+            status_bar,
+            station_catchment,
+        };
+
+        static widget_t widgets[] = {
+            commonWidgets(223, 136),
+            make_widget({ 3, 44 }, { 219, 80 }, widget_type::scrollview, 1, 2),
+            make_widget({ 3, 125 }, { 195, 10 }, widget_type::wt_13, 1),
+            make_widget({ 198, 44 }, { 24, 24 }, widget_type::wt_9, 1, image_ids::show_station_catchment, string_ids::station_catchment),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << station_catchment);
+
+        // 0x0048E7C0
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            self->widgets[widx::scrollview].right = self->width - 24;
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            self->widgets[widx::status_bar].top = self->height - 12;
+            self->widgets[widx::status_bar].bottom = self->height - 3;
+            self->widgets[widx::status_bar].right = self->width - 14;
+
+            self->widgets[widx::station_catchment].right = self->width - 2;
+            self->widgets[widx::station_catchment].left = self->width - 25;
+
+            common::repositionTabs(self);
```

Comment:
After repositioning the tabs `activated_widgets` needs to be updated dependant on _112C786 and window number.